### ISSUE TITLE
Semantics of `Resource::isMemento`

### ISSUE THREAD

**Issue by ajs6f** [trellis-ldp/trellis#89]
Under what circumstances is `Resource::isMemento` meant to return `true`? In particular, should all calls to `CassandraResourceService.get(IRI, Instant)` return `Resource`s whose `isMemento` methods return `true`?

**Comment by acoburn** [comment]
This depends a bit on the implementation. If the persistence backend supports versioning, then yes, a response to `ResourceService.get(IRI, Instant)` would return a `Resource` to which the `isMemento` would return `true`. But if the persistence layer does not support versioning, then you may just return an ordinary `Resource` to which the `isMemento` method returns `false`.

**Comment by ajs6f** [comment]
It's coming up for me because I am about to embark on impling versioning. So in a versioning backend, the `isMemento` on `Resource`s that were returned from (plain) `ResourceService.get(IRI)` should return `false` (every time) but the `isMemento` on `Resource`s that were returned from (fancy) `ResourceService.get(IRI, Instant)` should always return `true`, right?

**Comment by acoburn** [comment]
That is correct. You can see that the default implementation for this method returns `false`: https://github.com/trellis-ldp/trellis/blob/master/trellis-api/src/main/java/org/trellisldp/api/Resource.java#L150, so there's no need to do anything special for the ordinary resources.

**Comment by ajs6f** [comment]
Cool, I will send a PR with more Javadocs.
